src/ddmd/mtype.d

```diff
@@ -3075,6 +3075,13 @@ extern (C++) abstract class Type : RootObject
     {
         v.visit(this);
     }
+
+    final TypeFunction toTypeFunction()
+    {
+        if (ty != Tfunction)
+            assert(0);
```

Comment:
FWIW we already had two similar discussions
- for Phobos there's consensus that all `assert`'s should have an error message (e.g. [1](https://github.com/dlang/phobos/pull/4568), [2](https://github.com/dlang/phobos/pull/4602) ...)
- for DMD it has been agreed that [having a custom assert handler that points to bugzilla](https://github.com/dlang/dmd/pull/6103), would be _very_ user-friendly.
Unfortunately this PR went nowhere as all the `assert(0)`'s trigger SEGFAULTs in release mode ...